src/web/client/common/worker/webworker.js

```diff
@@ -163,20 +138,79 @@ async function loadContainer(config, swpId, entityInfo) {
                 });
-            } catch (error) {
-                await self.postMessage({
+            }
+        }
+
+        audience.on("memberRemoved", (clientId, member) => {
+            if (!existingMembers.get(member.additionalDetails.AadObjectId)) {
+                self.postMessage({
+                    type: "member-removed",
+                    userId: member.additionalDetails.AadObjectId,
+                });
+                self.postMessage({
+                    type: "telemetry-info",
+                    eventName: "webExtensionWebWorkerMemberRemovedSuccess",
+                    userId: member.additionalDetails.AadObjectId,
+                });
+            } else {
+                self.postMessage({
                     type: "telemetry-error",
-                    methodName: "webWorker valueChanged",
-                    errorMessage: error?.message,
-                    error: error,
+                    methodName: "webWorker memberRemoved",
+                    errorMessage: "Web Extension WebWorker Member Removed Failed",
                 });
             }
         });
+
+        selectionSharedMap.on("valueChanged", async (changed, local) => {
+            if (!initial) {
+                try {
+                    const user = getUserIdByConnectionId(changed.key);
+                    const userConnections = audience
+                        .getMembers()
+                        .get(user.userId).connections;
+
+                    const userEntityIdArray = [];
+
+                    const connectionIdInContainer = await map
+                        .get("selection")
+                        .get();
+
+                    userConnections.forEach((connection) => {
+                        userEntityIdArray.push(
+                            connectionIdInContainer.get(connection.id)
+                        );
+                    });
+
+                    // aadObjectId is the unique identifier for a user
+                    await self.postMessage({
+                        type: "client-data",
+                        userId: user.aadObjectId,
+                        userName: user.userName,
+                        containerId: swpId,
+                        entityId: userEntityIdArray,
+                    });
+
+                    await self.postMessage({
+                        type: "telemetry-info",
+                        eventName: "webExtensionWebWorkerGetUserIdByConnectionIdSuccess",
+                        userId: user.aadObjectId,
+                    });
+                } catch (error) {
+                    await self.postMessage({
+                        type: "telemetry-error",
+                        methodName: "webWorker valueChanged",
+                        errorMessage: error?.message,
+                        error: error,
+                    });
+                }
+            } else {
+                initial = false;
```

Comment:
What are we trying to do here with initial? Doesn't look optimal - can you discuss this approach once with someone in team?

Reply:
@BidishaMS, lets discuss this

Reply:
Reviewed and discussed different approach with @Ritik. He has implemented the same